src/java.naming/share/classes/com/sun/jndi/ldap/sasl/LdapSasl.java

```diff
@@ -123,15 +125,23 @@ public static LdapResult saslBind(LdapClient clnt, Connection conn,
         try {
             // Prepare TLS Channel Binding data
             if (conn.isTlsConnection()) {
-                TlsChannelBindingType cbType =
-                        TlsChannelBinding.parseType(
+                TlsChannelBindingType cbType;
+                try {
+                    cbType = TlsChannelBinding.parseType(
                                 (String)env.get(TlsChannelBinding.CHANNEL_BINDING_TYPE));
+                } catch (ChannelBindingException e) {
+                    throw new SaslException(e.getMessage(), e);
+                }
                 if (cbType == TlsChannelBindingType.TLS_SERVER_END_POINT) {
                     // set tls-server-end-point channel binding
                     X509Certificate cert = conn.getTlsServerCertificate();
                     if (cert != null) {
-                        TlsChannelBinding tlsCB =
-                                TlsChannelBinding.create(cert);
+                        TlsChannelBinding tlsCB;
+                        try {
+                            tlsCB = TlsChannelBinding.create(cert);
+                        } catch (ChannelBindingException e) {
+                            throw new SaslException(e.getMessage());
```

Comment:
Why is there a difference compared to line 133?

Reply:
Right, that was a mistake.